tokio/src/executor/task/mod.rs

```diff
@@ -99,8 +110,14 @@ impl<S: 'static> Task<S> {
 
 impl<S: Schedule> Task<S> {
     /// Returns `self` when the task needs to be immediately re-scheduled
-    pub(crate) fn run(self, executor: &mut dyn FnMut() -> Option<NonNull<S>>) -> Option<Self> {
-        if unsafe { self.raw.poll(executor) } {
+    pub(crate) fn run<F>(self, executor: &mut F) -> Option<Self>
```

Comment:
Now that this is generic, we don't need the `&mut` any more either I don't think?